contributing/code/bc.rst

```diff
@@ -267,6 +275,11 @@ Turn static into non static                         No
 .. [4] When changing the parent class, the original parent class must remain an
        ancestor of the class.
 
+.. [5] The value of a constant may only be changed when the constants aren't
+       used in configuration (e.g. Yaml and XML files), as these do not support
+       constants and have to hardcode the value. For instance, event name
+       constants can't change the value without introducing a BC break.
```

Comment:
I think we should simply add:

>
Additionally, if a constant will likely be used in objects that are serialized, the value of
a constant should not be changed.

To your point Wouter, technically any constant could be used in a configuration file too - but there are a few that are very commonly used (i.e. it's a "main supported use-case") - and it's those that we should avoid).